extensions/amp-fx-collection/0.1/providers/fx-provider.js

```diff
@@ -110,6 +118,19 @@ export class FxElement {
 
     /** @private {number} */
     this.factor_ = parseFloat(element.getAttribute('data-parallax-factor'));
+
```

Comment:
What worries me about this style is that the specific FX are being scattered in more than one place.

For generic properties, like `easing` or `duration`, `data-duration` (without the preset name) should work fine. If they are to be invalid for certain presets, we can add validation rules.